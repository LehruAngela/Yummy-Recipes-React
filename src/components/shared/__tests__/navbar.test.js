import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Navbar from '../navbar';

const mockStore = configureStore();

let store;
let wrapper;
const initialState = {
  auth: {
    username: 'TestUser'
  }
};

describe('Navbar Component', () => {
  beforeEach(() => {
    store = mockStore(initialState)
    wrapper = shallow(<Navbar store={store} />)
  });

  it('renders without crashing', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
