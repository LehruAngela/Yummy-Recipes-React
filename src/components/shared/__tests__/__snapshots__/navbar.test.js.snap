// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component renders without crashing 1`] = `
<ContextProvider
  value={null}
>
  <Navbar
    getUsername={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    username="TestUser"
  />
</ContextProvider>
`;
